File: EzHome/app/src/main/java/com/comslin/ezhome/oriUi/http/function/RoomHttp.java
package com.comslin.ezhome.oriUi.http.function;

import com.comslin.ezhome.oriUi.http.bean.room.AddDevices;
import com.comslin.ezhome.oriUi.http.bean.room.DeleteDevice;
import com.comslin.ezhome.oriUi.http.bean.room.RoomAdd;
import com.comslin.ezhome.oriUi.http.bean.room.RoomUpdate;
import com.zhy.http.okhttp.request.RequestCall;

import static com.comslin.ezhome.oriUi.constant.HttpConstant.URL;

/**
 * Created by linChao on 2017-04-25.
 */

public class RoomHttp extends BaseHttpFunc {
    private static final String ADD_ROOM = URL + "/api/room/add";
    private static final String ADD_DEVICE = URL + "/api/room/addDevice";
    private static final String DELETE = URL + "/api/room/delete";
    private static final String DELETE_DEVICE = URL + "/api/room/deleteDevice";
    private static final String LIST = URL + "/api/room/list";
    private static final String UPDATE = URL + "/api/room/update";

    public static RequestCall addRoom(RoomAdd rq) {
        return getTokenBuilder(ADD_ROOM, rq);
    }

    public static RequestCall addDevice(AddDevices rq) {
        return getTokenBuilder(ADD_DEVICE, rq);
    }

    public static RequestCall delete() {
        return getTokenBuilder(DELETE, new Object());
    }

    public static RequestCall delete_device(DeleteDevice rq) {
        return getTokenBuilder(DELETE_DEVICE, rq);
    }

    public static RequestCall list() {
        return getTokenBuilder(LIST, new Object());
    }

    public static RequestCall update(RoomUpdate rq) {
        return getTokenBuilder(UPDATE, rq);
    }


}

File: EzHome/app/src/main/java/com/comslin/ezhome/oriUi/http/function/RoomHttp.kt
package com.comslin.ezhome.oriUi.http.function

import com.comslin.ezhome.oriUi.http.bean.room.AddDevices
import com.comslin.ezhome.oriUi.http.bean.room.DeleteDevice
import com.comslin.ezhome.oriUi.http.bean.room.RoomAdd
import com.comslin.ezhome.oriUi.http.bean.room.RoomUpdate
import com.comslin.ezhome.oriUi.http.function.HttpConstant.EzHomeHost
import com.zhy.http.okhttp.request.RequestCall

/**
 * Created by linChao on 2017-04-25.
 */

class RoomHttp : BaseHttpFunc() {
    companion object {
        private val ADD_ROOM = EzHomeHost + "/api/room/add"
        private val ADD_DEVICE = EzHomeHost + "/api/room/addDevice"
        private val DELETE = EzHomeHost + "/api/room/delete"
        private val DELETE_DEVICE = EzHomeHost + "/api/room/deleteDevice"
        private val LIST = EzHomeHost + "/api/room/list"
        private val UPDATE = EzHomeHost + "/api/room/update"

        fun addRoom(rq: RoomAdd): RequestCall {
            return BaseHttpFunc.getTokenBuilder(ADD_ROOM, rq)
        }

        fun addDevice(rq: AddDevices): RequestCall {
            return BaseHttpFunc.getTokenBuilder(ADD_DEVICE, rq)
        }

        fun delete(): RequestCall {
            return BaseHttpFunc.getTokenBuilder(DELETE, Any())
        }

        fun delete_device(rq: DeleteDevice): RequestCall {
            return BaseHttpFunc.getTokenBuilder(DELETE_DEVICE, rq)
        }

        fun list(): RequestCall {
            return BaseHttpFunc.getTokenBuilder(LIST, Any())
        }

        fun update(rq: RoomUpdate): RequestCall {
            return BaseHttpFunc.getTokenBuilder(UPDATE, rq)
        }
    }


}
